subscription.go

```diff
@@ -2,35 +2,19 @@ package pubsub
 
 import (
 	"context"
-	"github.com/libp2p/go-libp2p-core/peer"
-	"sync"
-)
-
-type EventType int
-
-const (
-	PeerJoin EventType = iota
-	PeerLeave
 )
 
+// Subscription handles the details of a particular Topic subscription.
+// There may be many subscriptions for a given Topic.
 type Subscription struct {
 	topic    string
 	ch       chan *Message
 	cancelCh chan<- *Subscription
-	err      error
 	ctx      context.Context
-
-	peerEvtCh chan PeerEvent
-	evtLogMx  sync.Mutex
-	evtLog    map[peer.ID]EventType
-	evtLogCh  chan struct{}
-}
-
-type PeerEvent struct {
-	Type EventType
-	Peer peer.ID
+	err      error
 }
 
+// Topic returns the topic string associated with the Subscription
 func (sub *Subscription) Topic() string {
```

Comment:
@vyzo no need to do this now, but wanted to point out that it might be worth deprecating this function and/or replacing it with one that returns a `Topic` instead of a string.